include/nbl/video/ILogicalDevice.h

```diff
@@ -147,7 +147,7 @@ class NBL_API2 ILogicalDevice : public core::IReferenceCounted, public IDeviceMe
         virtual IQueue::RESULT waitIdle() const = 0;
 
         //! Semaphore Stuff
-        virtual core::smart_refctd_ptr<ISemaphore> createSemaphore(const uint64_t initialValue) = 0;
+        virtual core::smart_refctd_ptr<ISemaphore> createSemaphore(ISemaphore::SCreationParams&& ) = 0;
```

Comment:
change signature to `(const uint64_t initialValue, ISemaphore::CreationParams&& = {})`